teamcity: failed test: TestHashJoinerAgainstProcessor

The following tests appear to have failed on master (testrace): TestHashJoinerAgainstProcessor

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestHashJoinerAgainstProcessor).

[#1471016](https://teamcity.cockroachdb.com/viewLog.html?buildId=1471016):


```
TestHashJoinerAgainstProcessor
--- FAIL: testrace/TestHashJoinerAgainstProcessor (2.610s)

------- Stdout: -------
--- join type = LEFT_OUTER onExpr = "" seed = 908130567229589334 run = 2 ---
--- lEqCols = [2] rEqCols = [2] ---
--- inputTypes = [{{StringFamily 0 0 [] 0x8aa5210 0 <nil> [] [] 1043 <nil>}} {{DecimalFamily 0 0 [] 0x8aa5210 0 <nil> [] [] 1700 <nil>}} {{BytesFamily 0 0 [] 0x8aa5210 0 <nil> [] [] 17 <nil>}}] ---
    columnar_operators_test.go:256: different results: no match found for row 2 of processor output
        processor output:
        		[['U_yNE' -479035563456471.5719 NULL NULL NULL NULL] [NULL -0.237520451036119866 NULL NULL NULL NULL] [e'\x7f\x10' 841003.510225546096 '\x' e'`\x1d\x0b\x1eq`Hg' 5.776233569115936899E+30 '\x'] [e'\tfE7r>g\x11<' -787.6949047628281566 '\x' e'`\x1d\x0b\x1eq`Hg' 5.776233569115936899E+30 '\x'] [e'\x7f\x10' 841003.510225546096 '\x' e'5(/g|\x14{lI' NaN '\x'] [e'\tfE7r>g\x11<' -787.6949047628281566 '\x' e'5(/g|\x14{lI' NaN '\x'] ['^B9Y' -8.325911500003506620E+32 '\x27' NULL NULL NULL] [e'}s\x1aK' 247782395.5945897822 '\xa4d810b9' NULL NULL NULL] ['' 3052845.04045367018 '\xaf303fe62a0860bc94' NULL NULL NULL] ['' 5.002904114281552886E+25 '\xcbece64bace0' NULL NULL NULL] [e'w+E\x1bU}>' 528769334596063960.8 '\x08f8ba' NULL NULL NULL] [e'n-Bj\fLp\\' -6.540112128244789014E+34 '\xf2def7' NULL NULL NULL]]
        columnar operator output:
        		[[e'\tfE7r>g\x11<' -832.5911500003506620 '\x' e'`\x1d\x0b\x1eq`Hg' 5.776233569115936899E+30 '\x'] [e'\x7f\x10' 2477823.955945897822 '\x' e'`\x1d\x0b\x1eq`Hg' 5.776233569115936899E+30 '\x'] [e'\tfE7r>g\x11<' -479.0355634564715719 '\x' e'5(/g|\x14{lI' NaN '\x'] [e'\x7f\x10' 305284.504045367018 '\x' e'5(/g|\x14{lI' NaN '\x'] ['^B9Y' -8.325911500003506620E+32 '\x27' NULL NULL NULL] [e'}s\x1aK' 247782395.5945897822 '\xa4d810b9' NULL NULL NULL] ['U_yNE' -479035563456471.5719 NULL NULL NULL NULL] ['' 3052845.04045367018 '\xaf303fe62a0860bc94' NULL NULL NULL] ['' 5.002904114281552886E+25 '\xcbece64bace0' NULL NULL NULL] [e'w+E\x1bU}>' 528769334596063960.8 '\x08f8ba' NULL NULL NULL] [NULL -0.237520451036119866 NULL NULL NULL NULL] [e'n-Bj\fLp\\' -6.540112128244789014E+34 '\xf2def7' NULL NULL NULL]]




```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/distsqlrun/columnar_utils_test.go
+++ b/pkg/sql/distsqlrun/columnar_utils_test.go
@@ -107,7 +107,7 @@ func verifyColOperator(
 	defer outProc.ConsumerClosed()
 	defer outColOp.ConsumerClosed()
 
-	var procRows, colOpRows sqlbase.EncDatumRows
+	var procRows, colOpRows []string
 	rowCount := 0
 	for {
 		rowProc, meta := outProc.Next()
@@ -129,15 +129,15 @@ func verifyColOperator(
 			break
 		}
 
+		expStr := rowProc.String(outputTypes)
+		retStr := rowColOp.String(outputTypes)
 		if anyOrder {
 			// We accumulate all the rows to be matched using set comparison when
 			// both "producers" are done.
-			procRows = append(procRows, rowProc.Copy())
-			colOpRows = append(colOpRows, rowColOp.Copy())
+			procRows = append(procRows, expStr)
+			colOpRows = append(colOpRows, retStr)
 		} else {
 			// anyOrder is false, so the result rows must match in the same order.
-			expStr := rowProc.String(outputTypes)
-			retStr := rowColOp.String(outputTypes)
 			if expStr != retStr {
 				return errors.Errorf("different results on row %d;\nexpected:\n   %s\ngot:\n   %s", rowCount, expStr, retStr)
 			}
@@ -147,14 +147,12 @@ func verifyColOperator(
 
 	if anyOrder {
 		used := make([]bool, len(colOpRows))
-		for i, procRow := range procRows {
+		for i, expStr := range procRows {
 			rowMatched := false
-			for j, colOpRow := range colOpRows {
+			for j, retStr := range colOpRows {
 				if used[j] {
 					continue
 				}
-				expStr := procRow.String(outputTypes)
-				retStr := colOpRow.String(outputTypes)
 				if expStr == retStr {
 					rowMatched = true
 					used[j] = true
@@ -163,7 +161,7 @@ func verifyColOperator(
 			}
 			if !rowMatched {
 				return errors.Errorf("different results: no match found for row %d of processor output\n"+
-					"processor output:\n		%s\ncolumnar operator output:\n		%s", i, procRows.String(outputTypes), colOpRows.String(outputTypes))
+					"processor output:\n		%v\ncolumnar operator output:\n		%v", i, procRows, colOpRows)
 			}
 		}
 		// Note: we do not check whether used is all true here because procRows and

